File: Leetcoding-Actions/Explore-Learn/Arrays/removeDuplicates.java
class Solution {
    public int removeDuplicates(int[] nums) {
        /**
        * T: O(n) and S: O(1)
        */
        int n = nums.length;
        if (n <= 1) return n;
        
        int i = 0;
        int currVal = nums[i];
        for (int j = 1; j < n; j++) {
            if (nums[j] != currVal) {
                nums[i+1] = nums[j];
                i++;
                currVal = nums[i];
            }
        }
        return i + 1;
    }
}

File: Leetcoding-Actions/Explore-Learn/Arrays/removeDuplicates.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        '''
        T: O(n) and S: O(1)
        '''
        if len(nums) <= 1: return len(nums)
        
        i = 0
        currVal = nums[i]
        for j in range(1, len(nums)):
            if nums[j] != currVal:
                i += 1
                nums[i] = nums[j]
                currVal = nums[i]
        return i + 1
